tests/src/tests/test_utils.go

```diff
@@ -0,0 +1,324 @@
+}
+
+func GetBinaryDecision() bool {
+	return GetRandomDecision(1, 1)
+}
+
+func GetRandomDecision(success int, failure int) bool {
+	setRandSeed()
+	return rand.Intn(success+failure) > failure-1
+}
+
+func GetRandomNumber(max int) int {
+	setRandSeed()
+	return rand.Intn(max) + 1
+}
+
+func GetDevfile(fileName string) TestDevfile {
+	testDevfile := TestDevfile{}
+	testDevfile.SchemaDevFile = schema.Devfile{}
+	testDevfile.FileName = fileName
+	testDevfile.SchemaDevFile.SchemaVersion = "2.0.0"
+	testDevfile.SchemaParsed = false
+	return testDevfile
+}
+
+func (devfile *TestDevfile) CreateDevfile(useParser bool) error {
+	var err error
+
+	fileName := devfile.FileName
+	if !strings.HasSuffix(fileName, ".yaml") {
+		fileName += ".yaml"
+	}
+
+	if useParser {
+		LogMessage(fmt.Sprintf("   .......use Parser to write devfile %s", fileName))
+		newDevfile, err := devfileData.NewDevfileData(devfile.SchemaDevFile.SchemaVersion)
+		if err != nil {
+			LogMessage(fmt.Sprintf(" ..... ERROR: creating new devfile : %v", err))
+		} else {
+			newDevfile.SetSchemaVersion(devfile.SchemaDevFile.SchemaVersion)
+
+			// add the commands to new devfile
+			for _, command := range devfile.SchemaDevFile.Commands {
+				newDevfile.AddCommands(command)
+			}
+			// add components to the new devfile
+			newDevfile.AddComponents(devfile.SchemaDevFile.Components)
+
+			ctx := devfileCtx.NewDevfileCtx(fileName)
+
+			err = ctx.SetAbsPath()
+			if err != nil {
+				LogMessage(fmt.Sprintf(" ..... ERROR: setting devfile path : %v", err))
+			} else {
+				devObj := parser.DevfileObj{
+					Ctx:  ctx,
+					Data: newDevfile,
+				}
+				err = devObj.WriteYamlDevfile()
+				if err != nil {
+					LogMessage(fmt.Sprintf(" ..... ERROR: wriring devfile : %v", err))
+				}
+			}
+
+		}
+	} else {
+		LogMessage(fmt.Sprintf("   .......marshall and write devfile %s", devfile.FileName))
+		c, err := yaml.Marshal(&(devfile.SchemaDevFile))
+
+		if err == nil {
+			err = ioutil.WriteFile(fileName, c, 0644)
+		}
+	}
+	if err == nil {
+		devfile.SchemaParsed = false
+	}
+	return err
+}
+
+func (devfile *TestDevfile) ParseSchema() error {
+
+	var err error
+	if !devfile.SchemaParsed {
+		LogMessage(fmt.Sprintf(" -> Parse and Validate %s : ", devfile.FileName))
+		devfile.ParsedSchemaObj, err = devfilepkg.ParseAndValidate(devfile.FileName)
+		if err != nil {
+			LogMessage(fmt.Sprintf(" ......ERROR from ParseAndValidate %v : ", err))
+		}
+		devfile.SchemaParsed = true
+	}
+	return err
+}
+
+func (devfile TestDevfile) Verify() error {
+
+	LogMessage(fmt.Sprintf("Verify %s : ", devfile.FileName))
+
+	err := devfile.ParseSchema()
+
+	if err == nil {
+		LogMessage(fmt.Sprintf(" -> Get commands %s : ", devfile.FileName))
+		commands, _ := devfile.ParsedSchemaObj.Data.GetCommands(common.DevfileOptions{})
+		if commands != nil && len(commands) > 0 {
+			err = devfile.VerifyCommands(commands)
+		} else {
+			LogMessage(fmt.Sprintf("  No command found in %s : ", devfile.FileName))
```

Comment:
```suggestion
			LogMessage(fmt.Sprintf("  No commands found in %s : ", devfile.FileName))
```

Reply:
lol, the patent attorney in me says that "command" is more accurate.